modules/raft/src/main/java/org/apache/ignite/raft/jraft/entity/codec/v1/V1Encoder.java

```diff
@@ -107,32 +183,69 @@ public byte[] encode(final LogEntry log) {
         return content;
     }
 
-    private static int nodesListSizeInBytes(@Nullable List<PeerId> nodes, List<String> nodeStrs) {
-        int size = 0;
-
+    private static int nodesListSizeInBytes(@Nullable List<PeerId> nodes) {
         if (nodes != null) {
+            int size = 0;
+
             for (PeerId node : nodes) {
-                String nodeStr = node.toString();
+                String consistentId = node.getConsistentId();
 
-                nodeStrs.add(nodeStr);
-                size += 2 + nodeStr.length();
+                size += 2 + consistentId.length() + sizeInBytes(node.getIdx()) + sizeInBytes(node.getPriority() + 1);
             }
+
+            return size + sizeInBytes(nodes.size());
         }
 
-        return size + sizeInBytes(nodeStrs.size());
+        return 1;
```

Comment:
How about inverting the `if` condition and returning `1` if `nodes` is null? This will make it obvious what 1 means here